draco/synthesis/stream.py

```diff
@@ -243,6 +246,162 @@ def process(self, sstream):
+
+    Requires an input dataset with a time axis to mimic.
+
+    Attributes
+    ----------
+    lanczos_width : int
+        Width of the Lanczos interpolation kernel.
+    """
+
+    lanczos_width = config.Property(proptype=int, default=5)
+
+    def setup(self, observer):
+        """Set up a telescope object.
+
+        Parameters
+        ----------
+        observer : ProductManager
+            Telescope manager
+        """
+        self.observer = io.get_telescope(observer)
+
+    def process(self, sstream, tstream):
+        """Sample the sidereal stream at a set of times.
+
+        Parameters
+        ----------
+        sstream : containers.SiderealStream | containers.HybridVisStream
+            Dataset sampled in RA
+        tstream : containers.TODContainer | HybridVisStream
+            Dataset sampled in time
+
+        Returns
+        -------
+        timestream : containers.TimeStream | containers.HybridVisStream
+            sstream data sampled in time
+        """
+        # Figure out the time-like axis of `tstream`
+        if hasattr(tstream, "time"):
+            time = tstream.time[:]
+            tra = self.observer.unix_to_lsa(time)
+        elif hasattr(tstream, "ra"):
+            tra = tstream.ra[:]
+            lsd = tstream.attrs.get("lsd", tstream.attrs.get("csd"))
+            time = self.observer.lsd_to_unix(lsd + tra / 360.0)
+
+        # Determine output container based on input container
+        container_map = {
+            containers.SiderealStream: containers.TimeStream,
+            containers.HybridVisStream: containers.HybridVisStream,
+        }
+
+        # Make the new timestream containers
+        for cls in inspect.getmro(sstream.__class__):
+            OutputContainer = container_map.get(cls)
+
+            if OutputContainer is not None:
+                break
+
+        if OutputContainer is None:
+            raise TypeError(
+                f"No valid container mapping.\nGot {sstream.__class__}.\n"
+                f"Mappings exist for {list(container_map.keys())}."
+            )
+
+        # Provide both the time and ra axes - whichever one isn't
+        # used will be ignored
+        out = OutputContainer(axes_from=sstream, attrs_from=sstream, time=time, ra=tra)
+
+        # Make sure datasets are distributed correctly
+        sstream.redistribute("freq")
+        out.redistribute("freq")
+
+        # Make the interpolation array
+        R = regrid.lanczos_forward_matrix(sstream.ra, tra, self.lanczos_width).T.copy()
+
+        # Move the RA axis to the back
+        axind = list(sstream.data.attrs["axis"]).index("ra")
+        data = np.moveaxis(sstream.data[:].local_array, axind, -1)
+
+        # Apply the interpolation matrix
+        data_ts = data @ R
+
+        # Move the time-like axis back to where it belongs
+        out.data[:].local_array[:] = np.moveaxis(data_ts, -1, axind)
+        # Just set the weights to one
+        out.weight[:].local_array[:] = 1.0
```

Comment:
I would probably just implement the weight propagation.  I think it would follow something like:
```
waxind = list(sstream.weight.attrs["axis"]).index("ra")
var = np.moveaxis(tools.invert_no_zero(sstream.weight[:].local_array), waxind, -1)
out.weight[:].local_array[:] = np.moveaxis(tools.invert_no_zero(var @ R ** 2), -1, waxind)
```